lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

```diff
@@ -120,11 +199,25 @@ def main():
     # VM already exists
     if vm:
         try:
-            module.exit_json(instance=pyv.gather_facts(vm))
+            instance = pyv.gather_facts(vm)
+            if module.params.get('tags'):
+                if not HAS_VCLOUD:
+                    module.fail_json(msg="Unable to find 'vCloud Suite SDK' Python library which is required."
+                                         " Please refer this URL for installation steps"
+                                         " - https://code.vmware.com/web/sdk/60/vcloudsuite-python")
+
+                vm_rest_client = VmwareTag(module)
+                instance.update(
+                    tags=vm_rest_client.get_vm_tags(vm_rest_client.tag_service,
```

Comment:
The latest pyVmomi 6.7.1 seems to have tag support without going to REST.  I only have a VCSA in my environment (v6.7).  Perhaps we just need to update pyVmomi and don't need the vCloud / REST requirement here.

When I serialize out the contents of a virtual machine I see this now:
```
    "tag": [
        {
            "dynamicProperty": [], 
            "dynamicType": null, 
            "key": "SYSTEM/COM.VMWARE.VIM.VC"
        }
    ], 
```

Reply:
OK, does it work for 6.5 and Pyvmomi 6.5 ? I was getting 
```
"tag": []
```

Reply:
I just tried it with pyVmomi 6.7.1, not sure about any other versions.  Would need to check the pyVmomi project commits to see what's different.